common/models/model-method.js

```diff
@@ -10,4 +12,53 @@ module.exports = function(ModelMethod) {
    * @class ModelMethod
    * @inherits WorkspaceEntity
    */
+  app.once('ready', function() {
+    ready(ModelMethod);
+  });
 };
+
+
+function ready(ModelMethod) {
+  var debug = require('debug')('workspace:model-method');
+  var PackageDefinition = app.models.PackageDefinition;
+  var ModelDefinition = app.models.ModelDefinition;
+  
+  PackageDefinition.observe('after save', function getPackageDef(ctx, next) {
+    var pkg = ctx.instance ? ctx.instance.toObject() : ctx.data;
+
+    if (!pkg) {
+      debug('Cannot find Loopback version, defaulting to 2.x.');
+    } else {
+      var version = pkg.dependencies['loopback'];
+      var range = semver.Range(version);
+       // pkg.devDependencies['loopback'] = '1.x';
+      // pkg.optionalDependencies['loopback'] = '1.x';
+      debug('Loopback version: %s', version);
+    
+      if (semver.satisfies('3.0.0', range)) {
```

Comment:
1) I think this will not work, what if the range is `^3.10.0`? The value "3.0.0" will not satisfy that range.

2) Please move this code to a method in `PackageDefinition`, e.g. `PackageDefinition.getLoopBackVersion()`.